egs2/librispeech/hubert_asr1/hubert_asr.sh

```diff
@@ -0,0 +1,552 @@
+
+    # [Task dependent] Set the datadir name created by local/data.sh
+    --pretrain_train_set     # Name of pretraining train set
+    --pretrain_valid_set     # Name of pretraining valid set
+    --finetune_train_set     # Name of training set (required).
+    --finetune_valid_set     # Name of validation set used for monitoring/tuning network training (required).
+    --finetune_test_sets     # Name of test sets used for evaluating network training (required).
+                    # Multiple items (e.g., both dev and eval sets) can be specified (required).
+
+    --bpe_train_text # Text file path of bpe training set.
+    --lm_train_text  # Text file path of language model training set.
+    --lm_dev_text   # Text file path of language model development set (default="${lm_dev_text}").
+    --lm_test_text  # Text file path of language model evaluation set (default="${lm_test_text}").
+    --nlsyms_txt    # Non-linguistic symbol list if existing (default="${nlsyms_txt}").
+    --cleaner       # Text cleaner (default="${cleaner}").
+    --g2p           # g2p method (default="${g2p}").
+    --lang          # The language type of corpus (default=${lang}).
+    --score_opts             # The options given to sclite scoring (default="{score_opts}").
+    --local_score_opts       # The options given to local/score.sh (default="{local_score_opts}").
+    --asr_speech_fold_length # fold_length for speech data during ASR training (default="${asr_speech_fold_length}").
+    --asr_text_fold_length   # fold_length for text data during ASR training (default="${asr_text_fold_length}").
+    --lm_fold_length         # fold_length for LM training (default="${lm_fold_length}").
+EOF
+)
+
+log "$0 $*"
+# Save command line args for logging (they will be lost after utils/parse_options.sh)
+run_args=$(pyscripts/utils/print_args.py $0 "$@")
+. utils/parse_options.sh
+
+if [ $# -ne 0 ]; then
+    log "${help_message}"
+    log "Error: No positional arguments are required."
+    exit 2
+fi
+
+. ./path.sh
+. ./cmd.sh
+
+# Check required arguments
+[ -z "${finetune_train_set}" ] && { log "${help_message}"; log "Error: --finetune_train_set is required"; exit 2; };
+[ -z "${finetune_valid_set}" ] && { log "${help_message}"; log "Error: --finetune_valid_set is required"; exit 2; };
+[ -z "${finetune_test_sets}" ] && { log "${help_message}"; log "Error: --finetune_test_sets is required"; exit 2; };
+
+# Check feature type
+if [ "${feats_type}" = raw ]; then
+    data_feats=${dumpdir}/raw
+elif [ "${feats_type}" = fbank_pitch ]; then
+    data_feats=${dumpdir}/fbank_pitch
+elif [ "${feats_type}" = fbank ]; then
+    data_feats=${dumpdir}/fbank
+elif [ "${feats_type}" == extracted ]; then
+    data_feats=${dumpdir}/extracted
+else
+    log "${help_message}"
+    log "Error: not supported: --feats_type ${feats_type}"
+    exit 2
+fi
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    log "Stage 1: Data preparation for Librispeech & Librilight"
```

Comment:
This stage should be common.
```suggestion
    log "Stage 1: Data preparation"
```